src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs

```diff
@@ -53,6 +53,10 @@ public static ValueUsageInfo GetValueUsageInfo(this IOperation operation, ISymbo
                 // Declaration expression is a definition (write) for the declared local.
                 return ValueUsageInfo.Write;
             }
+            else if (operation is IRecursivePatternOperation or IListPatternOperation)
```

Comment:
Should we check if operation.DeclaredSymbol is non-null?

Reply:
@mavasani Does this apply to declaration pattern check below too?

Reply:
I am not sure it matters for either case, I was just thinking out loud. Maybe we should add a debug assert for both cases that DeclaredSymbol is non-null given that we have the null check in the operation walker. Is the GetValueUsageInfo API invoked anywhere apart from this analyzer? If not, it would be safe to add this assert.

Reply:
Probably we should return `ValueUsageInfo.None` in this case?

Reply:
>  Is the GetValueUsageInfo API invoked anywhere apart from this analyzer? 

It's, but for known operation types that don't involve patterns.